src/services/news.ts

```diff
@@ -0,0 +1,34 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import NewsOrComment from '../interfaces/NewsOrComment';
+import StoredNews from '../interfaces/StoredNews';
+
+export const newsApi = createApi({
+  reducerPath: 'newsApi',
+  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.hnpwa.com/v0/' }),
+  endpoints: (builder) => ({
+    getNewsOrCommentById: builder.query<NewsOrComment, number>({
+      query: (id) => {
+        return `item/${id}.json`;
+      },
+    }),
+    getNewest: builder.query<StoredNews[], number>({
+      async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
+        let pageNum = 1;
+        let newsCount = 0;
+        let results: StoredNews[] = [];
+        while (newsCount < _arg) {
```

Comment:
Опять же можешь не создавать отдельную переменную для хранения длины массива, а просто высчитывать её
```suggestion
        while (results.length< _arg) {
```